src/renderers/WebGLRenderer.js

```diff
@@ -1088,8 +1095,18 @@ function WebGLRenderer( parameters ) {
 
 		// _gl.finish();
 
+		currentRenderCallStack.pop();
+		if ( currentRenderCallStack.length > 0 ) {
```

Comment:
I wonder if it's not better to hide this logic in `WebGLRenderStates` (maybe in a new method `reset()`)? In this way, you could also move the stack into this class.

Reply:
`renderStates.get()` is called also in `compile()`. I don't think I'd hide this logic in `WebGLRenderStates`.

Reply:
I see.